thunderfish/eventdetection.py

```diff
@@ -145,6 +145,121 @@ def detect_peaks(data, threshold):
     return np.asarray(peaks_list, dtype=np.int), np.asarray(troughs_list, dtype=np.int)
 
 
+@jit(nopython=True)
+def detect_peaks_c(data, threshold):
```

Comment:
Why do you need both a jit and a non-jit version?

Reply:
Because the current jit version does not allow for threshold arrays. Functions in numba can only have one type, so having both floats and lists as inputs is (as far as I understood) not possible. I now only implemented the function for fixed threshold to be compiled with jit. I could do the same for arrays, but then I have to change all instances where this function is called. 

Furthermore, the function takes a bit to compile so for looking at single files using numba might even slow it down more. That's why I wanted to keep the option to not compile it.

Reply:
Ok, I got it! For numba you need a version for the scalar threshold and another one for the array threshold.

To make it run when numba is not installed, it might help to define the index type in the following way:
```
try:
    from numba import jit, int64
    index_type = int64
except ImportError:
    def jit(nopython):
        def decorator_jit(func):
            return func
        return decorator_jit
    index_type = np.int
```

and then

```
     peaks_list = np.zeros(len(data)//2,dtype=index_type)
```

And then one can have a function for the scalar threshold that uses numba if available and another one for the array threshold. Whatever, it is only important that it runs both with numba present and absent.